src/xmipp/libraries/classification/naive_bayes.cpp

```diff
@@ -339,6 +340,24 @@ NaiveBayes::~NaiveBayes()
     delete dummyLeaf;
 }
 
+// Assignment --------------------------------------------------------------
+NaiveBayes & NaiveBayes::operator=(const NaiveBayes &other)
+{
+	K=other.K;
+	Nfeatures=other.Nfeatures;
+    __priorProbsLog10=other.__priorProbsLog10;
+    __weights=other.__weights;
+    size_t imax=__leafs.size();
+    for (size_t i=0; i<imax; ++i)
+    	delete __leafs[i];
+    __leafs.clear();
```

Comment:
Why do you have a vector of pointers to leaves?
Wouldn't it be easier to have just vector of leaves?